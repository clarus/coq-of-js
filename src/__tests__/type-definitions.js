// @flow
import {compileAndPrint} from "../compiler/index.js";

it("handle type synonyms of constants", () => {
  expect(
    compileAndPrint(
      `type b = boolean;
type s = string;
type v = void;
type n = null;
type f = number;
type o = {};
type e = empty;
`,
    ),
  ).toMatchSnapshot();
});

it("handle type definition of enums", () => {
  expect(
    compileAndPrint(`type SayHi = "hi" | 'hello' | "h";`),
  ).toMatchSnapshot();

  expect(compileAndPrint('type Single = "s";')).toMatchSnapshot();
});

it("does not handle enums with other elements than strings", () => {
  expect(
    compileAndPrint(`type SayHi = "hi" | 'hello' | 12;`),
  ).toMatchSnapshot();
});

it("does not handle other literals than string", () => {
  expect(compileAndPrint(`type Boo = false;`)).toMatchSnapshot();

  expect(compileAndPrint(`type Num = 12;`)).toMatchSnapshot();
});
