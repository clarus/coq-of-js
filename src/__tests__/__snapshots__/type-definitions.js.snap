// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not handle enums with other elements than strings 1`] = `
"> 1 | type SayHi = \\"hi\\" | 'hello' | 12;
    |                              ^^

Only strings are handled in enums"
`;

exports[`does not handle other literals than string 1`] = `
"> 1 | type Boo = false;
    |           ^^^^^

Boolean literals in types are not handled"
`;

exports[`does not handle other literals than string 2`] = `
"> 1 | type Num = 12;
    |           ^^

Number literals in types are not handled"
`;

exports[`handle type definition of enums 1`] = `
"Inductive SayHi : Type :=
| hi
| hello
| h."
`;

exports[`handle type definition of enums 2`] = `
"Inductive Single : Type :=
| s."
`;

exports[`handle type synonyms of constants 1`] = `
"Definition b : Type := bool.

Definition s : Type := string.

Definition v : Type := unit.

Definition n : Type := unit.

Definition f : Type := Z.

Definition o : Type := unit.

Definition e : Type := Empty_set."
`;
